package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -92,3 +93,74 @@ def __init__(self,
     def _transform(self, ts):
         ts.dimensions = self.dimensions
         return ts
+
+
+class set_variable_dimensions(TransformationBase):
```

Comment:
Could we just move this new functionality as a new option of the existing `set_dimension`? The order of how things are done is very similar and it would be a much better user experience to not have to deal with (choose between) a lot of classes that do very similar things.